### PR TITLE
Add PHP 8-only support (v2)

### PR DESCRIPTION
**Pull request by patinthehat**
This PR adds a new major version, v2.0.0.

Specifically, it:

- Removes support for all PHP 7.x versions.
- Requires PHP 8.0+.
- All syntax converted to PHP 8 where possible.
- Removes unnecessary PHP docblocks per Spatie's guidelines.
- Moves exception classes to an "Exceptions" directory to match the structure of other packages.
- Adds config & github workflow for `php-cs-fixer`.
- Updates the location of `CONTRIBUTING.md`.
- Updates the readme with minor adjustments from [spatie/package-skeleton-php](https://github.com/spatie/package-skeleton-php).
- _Updates the changelog - release date needs to be updated, currently is "unreleased"._

### CODE REVIEW

File: .php_cs.cache [hunk 1]
```diff
@@ -0,0 +1 @@
...of":true,"single_class_element_per_statement":{"elements":["property"]},"single_import_per_statement":true,"single_line_after_imports":true,"switch_case_semicolon_to_colon":true,"switch_case_space":true,"visibility_required":true,"encoding":true,"full_opening_tag":true,"array_syntax":{"syntax":"short"},"ordered_imports":{"sortAlgorithm":"alpha"},"no_unused_imports":true,"not_operator_with_successor_space":true,"trailing_comma_in_multiline_array":true,"phpdoc_scalar":true,"unary_operator_spaces":true,"binary_operator_spaces":true,"blank_line_before_statement":{"statements":["break","continue","declare","return","throw","try"]},"phpdoc_single_line_var_spacing":true,"phpdoc_var_without_name":true,"class_attributes_separation":{"elements":["method"]},"single_trait_insert_per_statement":true},"hashes":{"src\/ReplaceResult.php":2133464100,"src\/Exceptions\/RegexFailed.php":3160576646,"src\/Regex.php":2371007970,"src\/MatchResult.php":1405433389,"src\/RegexResult.php":796781933,"src\/MatchAllResult.php":2786632830,"src\/Helpers\/Arr.php":1441465581,"src\/Helpers\/Str.php":2219256448,"tests\/ReplaceTest.php":2375454702,"tests\/MatchTest.php":463072247,"tests\/MatchAllTest.php":3755315586}}
```
Comment (freekmurze) [inline .php_cs.cache:1]: This file should not be included in the PR.

File: src/MatchAllResult.php [hunk 2]
```diff
@@ -3,39 +3,21 @@
...
-    /** @var array */
-    protected $matches;
-
-    public function __construct(string $pattern, string $subject, bool $result, array $matches)
-    {
-        $this->pattern = $pattern;
-        $this->subject = $subject;
-        $this->hasMatch = $result;
-        $this->matches = $matches;
+    public function __construct(
```
Comment (freekmurze) [inline src/MatchAllResult.php:11]: Nice!

File: src/MatchResult.php [hunk 3]
```diff
@@ -56,20 +38,12 @@ public function hasMatch(): bool
         return $this->hasMatch;
     }
 
-    /**
-     * @return string|null
-     */
-    public function result()
+    public function result(): string | null
```
Comment (freekmurze) [inline src/MatchResult.php:41]: `?string` might be nicer to use.

**Review by freekmurze — CHANGES_REQUESTED** [review]

**Comment by patinthehat** [comment]
ping @freekmurze

**Comment by freekmurze** [comment]
Thanks!

**Pull request merged**